src/layouts/EditPage.jsx

```diff
@@ -100,165 +99,160 @@ const getBackButtonInfo = (resourceCategory, collectionName, siteName) => {
         );
+        console.log(err);
       }
-      console.log(error)
     }
-    
-    if (!content) return
-    
-    try {
-      // split the markdown into front matter and content
-      const { frontMatter, mdBody } = frontMatterParser(Base64.decode(content));
-      
-      // retrieve CSP
-      const cspResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/netlify-toml`);
-      const { netlifyTomlHeaderValues } = cspResp.data;
-      const csp = new Policy(netlifyTomlHeaderValues['Content-Security-Policy']);
-
-      let leftNavPages
-      if (this.props.isCollectionPage) {
-        const collectionsApiEndpoint = getCollectionsApiEndpoint(this.apiEndpoint)
-        const collectionPagesResp = await axios.get(collectionsApiEndpoint);
-        const collectionResp = collectionPagesResp.data?.collectionPages;
-
-        // Retrieve third_nav_title from collection pages
-        leftNavPages = await Bluebird.map(collectionResp, async (collectionPage) => {
-          const collectionPageResp = await axios.get(`${collectionsApiEndpoint}/pages/${collectionPage.fileName}`)
-          const { content } = collectionPageResp.data;
-          const { frontMatter } = frontMatterParser(Base64.decode(content));
-          return {
-            ...collectionPage,
-            third_nav_title: frontMatter.third_nav_title,
-          }
-        });
-      }
 
-      if (this._isMounted) this.setState({
-        csp,
-        sha,
-        originalMdValue: mdBody.trim(),
-        editorValue: mdBody.trim(),
-        frontMatter,
-        leftNavPages,
-        isLoadingPageContent: false,
-      });
-    } catch (err) {
-      toast(
-        <Toast notificationType='error' text={`There was a problem trying to load your page. ${DEFAULT_ERROR_TOAST_MSG}`}/>, 
-        {className: `${elementStyles.toastError} ${elementStyles.toastLong}`}
-      );
-      console.log(err);
+    loadPageDetails()
+    return () => {
+      _isMounted = false
     }
-  }
-
-  componentWillUnmount() {
-    this._isMounted = false;
-  }
+  }, [])
 
-  setSelectedFile = (selectedFile) => {
-    this.setState({ selectedFile })
-  }
+  useEffect(() => {
+    const html = marked(editorValue)
+    const { isCspViolation, sanitisedHtml } = checkCSP(csp, html)
+    const chunk = prependImageSrc(siteName, sanitisedHtml)
+    setIsCspViolation(isCspViolation)
+    setChunk(chunk)
+  }, [editorValue])
 
-  updatePage = async () => {
+  const updatePage = async () => {
     try {
-      const { state } = this;
-      const { editorValue, frontMatter } = state;
-
       // here, we need to re-add the front matter of the markdown file
       const upload = concatFrontMatterMdBody(frontMatter, editorValue);
 
       // encode to Base64 for github
       const base64Content = Base64.encode(upload);
       const params = {
         content: base64Content,
-        sha: state.sha,
+        sha: sha,
```

Comment:
nit: can just write `sha,` instead of `sha: sha`

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/337/commits/0b82a283d15f16cfaaafb0ffa5251f093030137d